PocketKit/Sources/Sync/PocketSource.swift

```diff
@@ -962,6 +981,16 @@ extension PocketSource {
         }
     }
 
+    public func archive(collectionStory: CollectionStory) {
+        space.performAndWait {
+            guard let collectionStory = space.backgroundObject(with: collectionStory.objectID) as? CollectionStory,
+                  let savedItem = collectionStory.item?.savedItem, savedItem.isArchived == false else {
```

Comment:
could we simplify and reuse the logic you added to `CollectionStory`?
```suggestion
                  collectionStory.isSaved
```